modules/ipc/methods/eth_sendTransaction.js

```diff
@@ -36,14 +36,16 @@ module.exports = class extends BaseProcessor {
             try {
                 _.each(payload.params[0], (val, key) => {
                     // if doesn't have hex then leave
-                    if (_.isString(val)) {
+                    if (_.isString(val) || _.isNumber(val)) {
 
                         // make sure all data is lowercase and has 0x
                         if (val) val = `0x${val.toLowerCase().replace('0x', '')}`;
 
                         if (val.match(/[^0-9a-fx]/igm)) {
```

Comment:
Why not be even stricter - this would allow arbitrary mix of hex-digits and `x`. Why not require

* Starts with `0x`
* Then must be `0-9a-fA-F`

Something like `/0x[0-9a-fA-F]/`